internal/services/apicenter/api_center_service_resource.go

```diff
@@ -0,0 +1,217 @@
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apicenter/2024-03-01/services"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ApiCenterServiceResource struct{}
+
+var _ sdk.ResourceWithUpdate = ApiCenterServiceResource{}
+
+type ApiCenterServiceResourceModel struct {
+	Name          string            `tfschema:"name"`
+	ResourceGroup string            `tfschema:"resource_group_name"`
+	Location      string            `tfschema:"location"`
+	Tags          map[string]string `tfschema:"tags"`
+}
+
+func (r ApiCenterServiceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ApiCenterServiceResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ApiCenterServiceResource) ModelObject() interface{} {
+	return &ApiCenterServiceResourceModel{}
+}
+
+func (r ApiCenterServiceResource) ResourceType() string {
+	return "azurerm_api_center_service"
+}
+
+func (r ApiCenterServiceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return services.ValidateServiceID
+}
+
+func (r ApiCenterServiceResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.ServicesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ApiCenterServiceResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			id := services.NewServiceID(subscriptionId, model.ResourceGroup, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing ApiCenter Service %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			expandedIdentity, err := identity.ExpandLegacySystemAndUserAssignedMap(metadata.ResourceData.Get("identity").([]interface{}))
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+
+			apiCenterService := services.Service{
+				Name:     &model.Name,
+				Location: model.Location,
```

Comment:
This should be normalised before being sent
```suggestion
				Location: location.Normalize(model.Location),
```